src/main/java/com/linkedin/kmf/common/ConfigDocumentationGenerator.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright 2016 LinkedIn Corp. Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.linkedin.kmf.common;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.Writer;
+import java.lang.reflect.Field;
+import org.apache.kafka.common.config.AbstractConfig;
+import org.apache.kafka.common.config.ConfigDef;
+
+
+/**
+ * Generates the table of configuration parameters, their documentation strings and default values.
+ */
+public class ConfigDocumentationGenerator {
+
+  private static void printHelp() {
+    System.out.println("ConfigDocumentationGenerator outputDirectory configClassNames...");
+  }
+
+  private static void printHtmlHeader(Writer out, String docClass) throws IOException {
+    out.write("<html><head><title>Kafka Monitoring Automatically Generated Documentation. </title></head><body>\n");
+    out.write("<h1>");
+    out.write(docClass);
+    out.write("</h1>\n");
+  }
+  private static void printHtmlFooter(Writer out) throws IOException {
+    out.write("</body>\n</html>\n");
+  }
+
+  public static void main(String[] argv) throws Exception {
+    if (argv.length < 2) {
+      printHelp();
+      System.exit(1);
+    }
+
+    File outputDir = new File(argv[0]);
+    if (!outputDir.exists()) {
+      outputDir.mkdirs();
+    }
+
+    for (int i = 1; i < argv.length; i++) {
```

Comment:
It seems that what you have done here is similar (if not the same) as the implementation of `org.apache.kafka.common.config.ConfigDef.toHtmlTable()`. Can we simply print the string `ConsumeServiceConfig.CONFIG.toHtmlTable()` to a html file for each config class instead of writing new code here to generate the html file? I tried both your patch and `System.out.println(ConsumeServiceConfig.CONFIG.toHtmlTable())`. They actually generate the same output (except the title).

More specifically, I think we can probably add the follow main methods to each subclass of `AbstractConfig` (e.g. `ConsumeServiceConfig`). Then we can invoke these main methods to generate html files.

```
public static void main(String[] args) {
  System.out.println(CONFIG.toHtmlTable());
}
```

Reply:
We can add this feature/revisit in a follow-up PR.